### ISSUE TITLE
7.x does not work with PHP 5.3.3

### ISSUE THREAD

**Issue by MadMind** [drush-ops/drush#544]
There is a bug in PHP 5.3.3 https://bugs.php.net/bug.php?id=52854 that causes drush to fail with the following error:

```
ReflectionException: Class Drush\Sql\Sql7 does not have a constructor,
so you cannot pass any constructor arguments in
ReflectionClass->newInstanceArgs()
(line 104 of drush/includes/drush.inc).
```

**Comment by heddn** [comment]
I'm getting the above error as well using the official RHEL PHP distribution. https://github.com/drush-ops/drush/issues/544#issuecomment-62539791 fixed the issue.

**Comment by chirale** [comment]
[diarmy solution](https://github.com/drush-ops/drush/issues/544#issuecomment-62630721) works fine in my case on php 5.3.3.

**Comment by jakeman2048** [comment]
https://github.com/drush-ops/drush/issues/544#issuecomment-62539791 worked for me as well.

Can we either have drush7 minimum required PHP version changed to 5.3.4+ or patched in order fully work with PHP 5.3.0+?

**Comment by norwegianblue-nz** [comment]
"Pretty sure this came up before and we are replying to folks that they should upgrade their PHP"
Poor answer. Not all users have the ability to upgrade their php. esp if on shared hosting for instance.
As there seems to be a soln (according to comments re a patch in this thread) why can it not by incorporated into the official code.

**Comment by taz77** [comment]
Not a poor answer. PHP 5.3 has been end-of-life for 10 months now. If you are on a host using 5.3 then I think it would be prudent to look for a new host because running anything end of life on the web is a bad idea - no security patches, no updates!  

https://secure.php.net/eol.php

**Comment by heddn** [comment]
While it isn't supported by php.net, redhat does support it on its LTS. Which is why it is important to me. My client wants to use RHEL.

**Comment by heddn** [comment]
I hadn't noticed, but I guess I'll need to use an older version of PHP. This is just one of the many fun things about supporting an old version of PHP.

<code>includes/preflight.inc:  define('DRUSH_MINIMUM_PHP', '5.4.5');</code>

**Comment by taz77** [comment]
Time to upgrade! :-)

**Comment by heddn** [comment]
I can't upgrade php but I can downgrade drush. These clients insist that a supported version of php because of RHEL  is a supported version. I personally think that using php 5.5 or 5.6 is a much better solution. But I gotta pick my battles.

**Comment by AndreBaumeier** [comment]
This issue still exists with PHP 5.3.3 and Drush 7.1.0. The fix https://github.com/drush-ops/drush/issues/544#issuecomment-62539791 still works as well.

A usual drush user will miss out any mysql related information in drush status output. executing drush sql-connect will then printout "exception 'ReflectionException' with message 'Class Drush\Sql\Sql7 does not have a constructor, so you cannot pass any constructor arguments".

Hopefully this will be indexed in google under some related keywords to save some time for others.

**Comment by brevilo** [comment]
Any ETA for a release containing the patch? Thanks.

**Comment by heddn** [comment]
@brevilo, there won't be one. You'll need to downgrade your version of drush. Drush 7.x has a minimum supported version of PHP 5.4.5

**Comment by brevilo** [comment]
So what's the point of #1838 then? Also, the [docs](http://www.drush.org/en/master/install/#drupal-compatibility) still say 5.3+. Downgrading to Drush 6 isn't an option since it's unmaintained and doesn't contain the SSL-update fix AFAIK...

**Comment by heddn** [comment]
Ahh, so, there's been an change in stance since #544. Maybe there is hope...

**Comment by dalin-** [comment]
I'm adding a patch file, just so that I have a URL of a patch so that I can build a version of drush7 that works on Debian LTS.  I don't actually expect anything to be committed, but anyone can also use this URL. 
[github-drush-ops-drush-544_drush_does_not_work_with_php_5.3.3.patch.txt](https://github.com/drush-ops/drush/files/184281/github-drush-ops-drush-544_drush_does_not_work_with_php_5.3.3.patch.txt)

**Comment by njt1982** [comment]
So after #1838 - does the 7.x simply need a new tagged release?

**Comment by AlexShapka** [comment]
we have several servers on which we can not upgrade php for now and which really needed drush 7. dalin's patch above worked like a charm. hopefully this will be committed. thanks!

**Comment by weitzman** [comment]
7.2.0 was released 29 days ago, and has the fix for this issue - #1838

**Comment by JorisVanEijden** [comment]
Unfortunately 7.2.0 also contains a composer.lock file that requires PHP >=5.3.9 [#2132]
So git+composer based installs won't work on 5.3.3 without manual fiddling.
